### ISSUE TITLE
External Database Server

### ISSUE THREAD

**Issue by SnowyJaguar1034** [boschkundendienst/guacamole-docker-compose#64]
Hi, Since my [last](#63) 'issue' I've been fairly busy so I've not had time to look into deploying your stack/compose but I've now got a bit more time. I've recently setup a postgres database container to act as a central database for all of my containers instead of each of them having their own.
I just have a few questions about deploying you stack/compose.
- Does the guacamole service create it's own database (`POSTGRES_DATABASE: guacamole`) or does the postgres service do that?
If it doesn't I'd have to prepare that ahead of time myself right? Could I put something in the compose to make it for me?
- If I comment-out the postgres service do I just complete the following variables with the following values?
```
environment:
      GUACD_HOSTNAME: guacd
      POSTGRES_DATABASE: guacamole (or whatever name I gave the database I created)
      POSTGRES_HOSTNAME: <IP_OF_MY_CENTRAL_DB_SERVER> e.g. 192.168.86.23:5432
      POSTGRES_PASSWORD: 'ChooseYourOwnPasswordHere1234'
      # POSTGRES_USER: guacamole  (or whatever name I gave the user of the database I created)
```
- As I'm not using nginx (I plan on using traefik at some point) and I'm not using the postgres service then I don't need to run `prepare.sh` right?

Sorry for creating issues for these questions, If you had the 'Disscussions' feature enabled then I would have asked there.

**Comment by SnowyJaguar1034** [comment]
Hi, Thank you for getting back to me.

I, unfortunately, didn't get my 'central' database idea to work due to permission issues I could not resolve.

So I instead just spun up the postgres instance in the stack. I have then been trying to access the postgres db from my pgadmin interface; however, I can't connect. I've already adjusted the stack to expose the postgres server on binding `5432:5432`; however, I can't connect.
I exec'd into the container to try to run some psql commands but I can't even connect to the db from within the container. I've tried `su `to get into the postgres user account but a 'role does not exist' error
```
root@ve:~# docker exec -it c01ca1b151ba bash
c01ca1b151ba:/# su postgres
/ $ psql
psql: error: connection to server on socket "/var/run/postgresql/.s.PGSQL.5432" failed: FATAL:  role "postgres" does not exist
/ $
``` 
I then tried the same command with `guacamole_user` but I get 'unknown user'. I then tried the following with no success
```
c01ca1b151ba:/var/lib/postgresql/data/guacamole# psql --host=localhost --port=5432 --username=postgres --password --dbname=guacamole
Password:
psql: error: connection to server at "localhost" (127.0.0.1), port 5432 failed: FATAL:  role "postgres" does not exist
c01ca1b151ba:/var/lib/postgresql/data/guacamole# psql --host=localhost --port=5432 --username=guacamole --password --dbname=guacamole
Password:
psql: error: connection to server at "localhost" (127.0.0.1), port 5432 failed: FATAL:  role "guacamole" does not exist
c01ca1b151ba:/var/lib/postgresql/data/guacamole#
```
I've had a good Google and looked at some of the common files people say to edit like `pg_hba.conf` and `postgresql.conf` and those files appear to suggest I should be able to access the db. I also checked the postgres service is actually running and it is.
Have you manged to connect to the db inn the container before? Is this question better suited to the guacamole team?

**Comment by boschkundendienst** [comment]
Hi, as far as I understand you want to use your own PostgreSQL instance. Well then you have to learn it. I just can show you in this video, that it is definitely possible to connect to the PostgreSQL database that runs within a container. Of course I took my setup as a reference. Maybe this helps? Anyways PostgreSQL is very tricky when it comes to versions. You can only handle a database with a specific version with the management tools that have the same version.

Take a [look here](https://asciinema.org/a/TyJ8AHFtViIGXP3XHXl2PD9Fc) and see that it works. (Asciinema  video)

I tested this a few minutes ago.

**Comment by SnowyJaguar1034** [comment]
Hi, just following up with this.

I originally wanted to use my own postgres instance and I did go down that path however I'd obviously got something wrong in the configuration somewhere because I was consistently getting a user permission (or possibly table permissions) error.

So once I'd spent a few hours trying to get this to work I had the idea to redeploy (all of) the containers from your compose (including your postgres) following your guidance and I got that to work (which was expected).

The stage I'm at now, and that my previous comment was referring to, is to make the postgres container from your compose accessible to a) primarily my pgadmin container to make the database easier to manage if that ever needs to happen and b) so that I can add my own databases into the database server.

Thanks for sending over that video, I'll take a look at it this weekend.

All of the issues I was having with accessing the psql cli ( the ones I showed code blocks for) were all for the postgres container included in your compose, I wasn't having those issues with the container I deployed separately. This may have been a configuration issue on my end, a configuration issue on your end or configuration issue from Apache Guacamole's images but I don't know enough to say definitively.

If your able to access the psql cli included in the postgres container from your compose then that would indicate it's an issue with my configuration however I did test deploy the compose 2 or 3 times and consistently had this issue so either I'm consistently making the same config issue or something else is going on.

Apologies for any confusion my previous comment caused about what I was saying and asking about.
